staging/src/k8s.io/apiserver/pkg/opentelemetry/config.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package opentelemetry
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/apis/apiserver"
+	"k8s.io/apiserver/pkg/apis/apiserver/install"
+	"k8s.io/apiserver/pkg/apis/apiserver/v1alpha1"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	cfgScheme = runtime.NewScheme()
+
+	defaultPort = int32(55680)
+	defaultURL  = "localhost:55680"
+)
+
+func init() {
+	install.Install(cfgScheme)
+}
+
+// ReadOpenTelemetryConfiguration reads the opentelemetry configuration from a file
+func ReadOpenTelemetryConfiguration(configFilePath string) (*apiserver.OpenTelemetryClientConfiguration, error) {
+	if configFilePath == "" {
+		return nil, fmt.Errorf("opentelemetry config file was empty")
+	}
+	data, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read opentelemetry configuration from %q [%v]", configFilePath, err)
+	}
+	var decodedConfig v1alpha1.OpenTelemetryClientConfiguration
+	err = yaml.Unmarshal(data, &decodedConfig)
+	if err != nil {
+		// we got an error where the decode wasn't related to a missing type
+		return nil, err
+	}
+	if decodedConfig.Kind != "OpenTelemetryClientConfiguration" {
+		return nil, fmt.Errorf("invalid service configuration object %q", decodedConfig.Kind)
+	}
+	internalConfig := &apiserver.OpenTelemetryClientConfiguration{}
+	if err := cfgScheme.Convert(&decodedConfig, internalConfig, nil); err != nil {
+		// we got an error where the decode wasn't related to a missing type
+		return nil, err
+	}
+	return internalConfig, nil
+}
+
+// DefaultOpenTelemetryConfiguration defaults unset fields in the OpenTelemetryClientConfiguration
+func DefaultOpenTelemetryConfiguration(config *apiserver.OpenTelemetryClientConfiguration) {
```

Comment:
if we use the regular decoding path these can be set in a normal API object defaulting function and will get set automatically at decode time

Reply:
done